### PR TITLE
Query string manager

### PR DESCRIPTION
**Pull request by mcarta-rt**
utility functions for query string api params.

### CODE REVIEW

File: src/util/queryStringManager.js [hunk 1]
```diff
@@ -11,18 +22,31 @@ const addFilter = (filterValues) => {
...
+
+export const setFilters = (object, operators = []) => {
   const properties = Object.keys(object);
-  const queryString = properties.map((property, index) => {
+  const filters = properties.map((property, index) => {
     const filterValues = {
       attribute: property,
       value: object[property],
-      operator,
+      operator: operators[property] ? operators[property] : DEFAULT_FILTER_OPERATOR,
```
Comment (nicpuddu) [inline src/util/queryStringManager.js]: use property in operators

File: test/util/queryStringManager.test.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  code: 'test_code',
+  value: 'test',
+  type: 'type',
+};
+
+const operators = {
+  code: 'eq',
+  value: 'eq',
+  type: 'eq',
+};
```
Comment (nicpuddu) [inline test/util/queryStringManager.test.js]: no magic string

File: test/util/queryStringManager.test.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    direction: 'ASC',
+  },
+};
+
+
+const QUERY_STRING_OK = '?sort=code&direction=ASC&filters[0][attribute]=code&filters[0]' +
+'[operator]=eq&filters[0][value]=test_code' +
+'&filters[1][attribute]=value&filters[1][operator]=eq' +
+'&filters[1][value]=test&filters[2][attribute]=type' +
+'&filters[2][operator]=eq&filters[2][value]=type';
```
Comment (nicpuddu) [inline test/util/queryStringManager.test.js]: escape new lines instead of concatenating

File: test/util/queryStringManager.test.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  it('test if properly generate attribute filters', () => {
+    const filter = addFilter(filterValues);
+    expect(filter).toEqual(FILTER_OK);
+  });
+});
+describe('test setSorting', () => {
+  it('test if properly generate a sort filter', () => {
+    const filter = setSorting(filters.sorting);
+    expect(filter.join('&')).toEqual(SORTING_OK);
+  });
```
Comment (nicpuddu) [inline test/util/queryStringManager.test.js]: test bad operator values

**Comment by entando-service** [comment]
ok to test

**Comment by entando-service** [comment]
add to whitelist

**Comment by entando-service** [comment]
ok to test

**Comment by entando-service** [comment]
retest this please

**Comment by entando-service** [comment]
add to whitelist

**Comment by entando-service** [comment]
test this please

**Comment by entando-service** [comment]
All is well

File: test/util/queryStringManager.test.js (hunk 4 above)
Comment (mcarta-rt) [reply test/util/queryStringManager.test.js]: Bad operators case tested on later in the test file.

**Pull request merged**